crates/driver/src/domain/competition/solution/encoding.rs

```diff
@@ -7,3 +22,330 @@ pub enum Strategy {
+                    buy_token_index: (tokens.len() - 1).into(),
+                    receiver: trade.order().receiver.unwrap_or_default().into(),
+                    sell_amount: trade.order().sell.amount.into(),
+                    buy_amount: trade.order().buy.amount.into(),
+                    valid_to: trade.order().valid_to.0.into(),
+                    app_data: trade.order().app_data.0 .0.into(),
+                    fee_amount: eth::U256::zero(),
+                    flags: order_flags(trade.order()),
+                    executed_amount: trade.executed().0,
+                    signature: trade.order().signature.data.clone(),
+                });
+
+                pre_interactions.extend(trade.order().pre_interactions.clone());
+                post_interactions.extend(trade.order().post_interactions.clone());
+            }
+            super::Trade::Jit(trade) => {
+                tokens.push(trade.order().sell.token.0.into());
+                tokens.push(trade.order().buy.token.0.into());
+
+                // Jit orders are matched at limit price, so the sell token is worth buy.amount
+                // and vice versa
+                clearing_prices.push(trade.order().buy.amount.into());
+                clearing_prices.push(trade.order().sell.amount.into());
+
+                trades.push(Trade {
+                    sell_token_index: (tokens.len() - 2).into(),
+                    buy_token_index: (tokens.len() - 1).into(),
+                    receiver: trade.order().receiver.into(),
+                    sell_amount: trade.order().sell.amount.into(),
+                    buy_amount: trade.order().buy.amount.into(),
+                    valid_to: trade.order().valid_to.0.into(),
+                    app_data: trade.order().app_data.0 .0.into(),
+                    fee_amount: eth::U256::zero(),
+                    flags: jit_order_flags(trade.order()),
+                    executed_amount: trade.executed().0,
+                    signature: trade.order().signature.data.clone(),
+                });
+            }
+        }
+    }
+
+    // Encode allowances
+    for approval in approvals {
+        interactions.push(approve(&approval.0))
+    }
+
+    // Encode interaction
+    for interaction in solution.interactions() {
+        if matches!(internalization, settlement::Internalization::Enable)
+            && interaction.internalize()
+        {
+            continue;
+        }
+
+        interactions.push(match interaction {
+            competition::solution::Interaction::Custom(interaction) => eth::Interaction {
+                value: interaction.value.into(),
+                target: interaction.target.0.into(),
+                call_data: interaction.call_data.clone(),
+            },
+            competition::solution::Interaction::Liquidity(interaction) => {
+                // Todo account for slippage
+                let input = MaxInput(interaction.input);
```

Comment:
I think moving the encoding into a separate function would help the readability of the encoding logic of the whole settlement. We already don't see any details on how the individual variants get encoded anyway.